captum/attr/_core/layer/layer_lrp.py

```diff
@@ -0,0 +1,218 @@
+            - **attributions** (*tensor* or tuple of *tensors*):
+                        The propagated relevance values with respect to each
+                        input feature. Attributions will always
+                        be the same size as the provided inputs, with each value
+                        providing the attribution of the corresponding input index.
+                        If a single tensor is provided as inputs, a single tensor is
+                        returned. If a tuple is provided for inputs, a tuple of
+                        corresponding sized tensors is returned. The sum of attributions
+                        is one and not corresponding to the prediction score as in other
+                        implementations.
+            - **delta** (*tensor* or list of *tensors* returned if
+                return_convergence_delta=True):
+
+                        Delta is calculated per example, meaning that the number of
+                        elements in returned delta tensor is equal to the number of
+                        of examples in input.
+                        If attributions for all layers are returned (layer=None) a list
+                        of tensors is returned with entries for each layer.
+        Examples::
+
+                >>> # ImageClassifier takes a single input tensor of images Nx3x32x32,
+                >>> # and returns an Nx10 tensor of class probabilities. It has one
+                >>> # Conv2D and a ReLU layer.
+                >>> net = ImageClassifier()
+                >>> lrp = LRP(net, net.conv1)
+                >>> input = torch.randn(3, 3, 32, 32)
+                >>> # Attribution size matches input size: 3x3x32x32
+                >>> attribution = lrp.attribute(input, target=5)
+
+        """
+        self.verbose = verbose
+        self._original_state_dict = self.model.state_dict()
+        self.layers = []
+        self._get_layers(self.model)
+        self._get_rules()
+        self.attribute_to_layer_input = attribute_to_layer_input
+        self.backward_handles = []
+        self.forward_handles = []
+
+        is_layer_tuple = self._is_layer_tuple(inputs, additional_forward_args)
+        inputs = _format_input(inputs)
+        gradient_mask = apply_gradient_requirements(inputs)
+        # 1. Forward pass
+        self._change_weights(inputs, additional_forward_args)
+        self._register_forward_hooks()
+        # 2. Forward pass + backward pass
+        _ = compute_gradients(
+            self._forward_fn_wrapper, inputs, target, additional_forward_args
+        )
+        relevances = self._get_output_relevance()
+
+        self._restore_model()
+        undo_gradient_requirements(inputs, gradient_mask)
+
+        if return_convergence_delta:
+            if self.layer is None:
+                delta = []
+                for relevance_layer in relevances[0]:
+                    delta.append(self.compute_convergence_delta(relevance_layer))
```

Comment:
`relevances` is always a tuple of length 1, I think to keep this tuple convention. Then in loops over the list of relevances, which will be only 1 for single layers. So, similar to the point above, we might need a different concept for relevances of intermediate layers with tuple inputs or outputs.

Reply:
Yeah, I think that returning relevances as `return (relevances,) ` in the `_get_output_relevance` is a bit artificial. I think that we can in case of `layer is None` we can always return a list or a tuple of layer relevances that are already correctly formatted in the `lrp_rules.py`. What do you think ?

Reply:
This was change in a previous commit already. `lrp_rules.py` returns a list of relevances for each layer, which is converted to a tuple in `lrp.py`.